core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -86,6 +87,8 @@ public void accept(String file) {
   private boolean stageOnly = false;
   private Consumer<String> deleteFunc = defaultDelete;
 
+  private ExecutorService planExecutorService = ThreadPools.getWorkerPool();
```

Comment:
What about calling this `workerPool` instead of executor service? I think that's a bit more clear and we use the term "thread pool" in most places.